packages/contrast-colors/index.js

```diff
@@ -279,6 +314,116 @@ function contrast(color, base) {
   }
 }
 
+function minPositive(r) {
+  if (!r) { throw new Error('Array undefined');}
+  let arr = [];
+
+  for(let i=0; i < r.length; i++) {
+    if(r[i] >= 1) {
+      arr.push(r[i]);
+    }
+  }
+  return Math.min(...arr);
+}
+
+function ratioName(r) {
+  if (!r) { throw new Error('Ratios undefined');}
+  r = r.sort(function(a, b){return a - b}); // sort ratio array in case unordered
+
+  let min = minPositive(r);
+  let minIndex = r.indexOf(min);
+  let nArr = []; // names array
+
+  let rNeg = r.slice(0, minIndex);
+  let rPos = r.slice(minIndex, r.length);
+
+  // Name the negative values
+  for (let i=0; i < rNeg.length; i++) {
+    let d = 1/(rNeg.length + 1);
+    let m = d * 100;
+    let nVal = m * (i + 1);
+    nArr.push(Number(nVal.toFixed()));
+  }
+  // Name the positive values
+  for (let i=0; i < rPos.length; i++) {
+    nArr.push((i+1)*100);
+  }
+  nArr.sort(function(a, b){return a - b}); // just for safe measure
+
+  return nArr;
+}
+
+function generateAdaptiveTheme({baseScale, colorScales, brightness, contrast = 1}) {
+  if (!baseScale) {
+    throw new Error(`baseScale is undefined`);
+  }
+  if (typeof baseScale !== 'object') {
+    throw new Error('baseScale must be an object')
+  }
+  if (!colorScales) {
+    throw new Error(`colorScales are undefined`);
+  }
+
+  if (brightness === undefined) {
+    return function(brightness, contrast) {
+      return generateAdaptiveTheme({baseScale: baseScale, colorScales: colorScales, brightness: brightness, contrast: contrast})
+    }
+  }
+  else {
+    let bscale = generateBaseScale(baseScale); // base parameter to create base scale (0-100)
+    let bval = bscale[brightness];
+    let arr = [];
+
+    for (let i=0; i < colorScales.length; i ++) {
```

Comment:
Nit: Prefer no space before `++`, prefer space before and after `=`. (we should add some linting)

Same issues elsewhere.